packages/openactive-integration-tests/test/features/leasing/anonymous-leasing/implemented/lease-opportunity-capacity-update-test.js

```diff
@@ -25,176 +26,91 @@ FeatureHelper.describeFeature(module, {
     control: false,
   }],
 },
-(configuration, orderItemCriteria, featureIsImplemented, logger, parentState, parentFlow) => {
-  /**
-   * @param {number} expected
-   * @param {C1} stage
-   * @param {() => ChakramResponse} responseAccessor
-   */
-  function itShouldHaveCapacity(expected, stage, responseAccessor) {
-    Common.itForOrderItem(orderItemCriteria, parentState, stage, () => responseAccessor().body,
-      'should decrement remaining slots',
-      (feedOrderItem, responseOrderItem) => {
-        if (responseOrderItem && responseOrderItem.orderedItem['@type'] === 'Slot') {
-          expect(responseOrderItem).to.nested.include({
-            'orderedItem.remainingUses': expected,
-          });
-        } else {
-          expect(responseOrderItem).to.nested.include({
-            'orderedItem.remainingAttendeeCapacity': expected,
-          });
-        }
-      });
-  }
+(configuration, orderItemCriteriaList, featureIsImplemented, logger) => {
+  // # First, get the Opportunity Feed Items which will be used in subsequent tests
+  const fetchOpportunities = new FetchOpportunitiesFlowStage({
+    ...FlowStageUtils.createSimpleDefaultFlowStageParams({ logger }),
+    orderItemCriteriaList,
+  });
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
 
   /**
-   * @param {C1} stage
-   * @param {() => ChakramResponse} responseAccessor This is wrapped in a
-   *   function because the actual response won't be available until the
-   *   asynchronous before() block has completed.
+   * @param {object} args
+   * @param {UnknownFlowStageType} args.prerequisiteFlowStage
+   * @param {number} args.numberOfItems How many of each OrderItem are we going to include in our lease?
+   * @param {number} args.expectedCapacityFromPreviousSuccessfulLeases What do we expect the Booking System to report
+   *   as the capacity in its response?
+   * @param {boolean} args.shouldSucceed Should this batch of leases succeed?
+   * @param {(c1: C1FlowStageType) => void} [itAdditionalTests]
    */
-  function itShouldReturn409Conflict(stage, responseAccessor) {
-    it('should return 409', () => {
-      stage.expectResponseReceived();
-      chakram.expect(responseAccessor()).to.have.status(409);
+  const describeNewLeaseOfXItems = ({
```

Comment:
I made this new function which generalises the pattern being used for each new lease of X items in this test